tests/alpaka/alpaka_main.cpp

```diff
@@ -0,0 +1,97 @@
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Alpaka include(s).
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <iostream>
+
+template <typename Acc>
+ALPAKA_FN_ACC float process(Acc const& acc, uint32_t idx) {
+    return alpaka::math::sin(acc, idx);
+}
+
+struct VectorOpKernel {
+    template <typename Acc>
+    ALPAKA_FN_ACC void operator()(Acc const& acc, float* result,
+                                  uint32_t n) const {
+        using namespace alpaka;
+        auto const globalThreadIdx =
+            getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
+
+        // It is possible to get index type from the accelerator,
+        // but for simplicity we just repeat the type here
+        if (globalThreadIdx < n) {
+            result[globalThreadIdx] = process(acc, globalThreadIdx);
+        }
+    }
+};
+
+/// Copy vector to device, set element[i] = sin(i) on device, then compare with
+/// same calculation on host
+GTEST_TEST(AlpakaBasic, VectorOp) {
+    using namespace alpaka;
+    using Dim = DimInt<1>;
+    using Idx = uint32_t;
+
+    using Acc = ExampleDefaultAcc<Dim, Idx>;
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<Acc>()
+              << std::endl;
+
+    // Select a device and create queue for it
+    auto const devAcc = getDevByIdx<Acc>(0u);
+
+    using Queue = Queue<Acc, Blocking>;
+    auto queue = Queue{devAcc};
+
+    uint32_t n = 10000;
+
+    uint32_t blocksPerGrid = n;
+    uint32_t threadsPerBlock = 1;
+    uint32_t elementsPerThread = 4;
+    using WorkDiv = WorkDivMembers<Dim, Idx>;
+    auto workDiv = WorkDiv{blocksPerGrid, threadsPerBlock, elementsPerThread};
+
+    // Create a device for host for memory allocation, using the first CPU
+    // available
+    auto devHost = getDevByIdx<DevCpu>(0u);
+
+    // Allocate memory on the device side:
+    auto bufAcc = alpaka::allocBuf<float, uint32_t>(devAcc, n);
+
+    alpaka::exec<Acc>(queue, workDiv, VectorOpKernel{},
+                      alpaka::getPtrNative(bufAcc), n);
+
+    alpaka::wait(queue);
+
+    // Allocate memory on the host side:
+    auto bufHost = alpaka::allocBuf<float, uint32_t>(devHost, n);
+    // Copy bufAcc to bufHost
+    alpaka::memcpy(queue, bufHost, bufAcc);
+    // Calculate on the host and compare result
+    for (uint32_t i = 0u; i < n; i++) {
+        bool testPassed = std::abs(bufHost[i] - std::sin(i)) < 1e-3;
+        if (!testPassed) {
+            std::cout << "bufHost[" << i << "]: " << bufHost[i] << std::endl;
+            std::cout << "bufAcc[" << i << "]: " << bufAcc[i] << std::endl;
+            std::cout << "std::sin(" << i << "): " << std::sin(i) << std::endl;
+        }
+        ASSERT_TRUE(testPassed);
+    }
+}
+
+int main(int argc, char** argv) {
+    testing::InitGoogleTest(&argc, argv);
+    int r = RUN_ALL_TESTS();
+    return r;
+}
```

Comment:
You shouldn't have a `main()` function in here. Instead, just add `GTest::gtest_main` to the libraries linked against the test. I.e.

```diff
diff --git a/tests/alpaka/CMakeLists.txt b/tests/alpaka/CMakeLists.txt
index f581cca..e44dfa4 100644
--- a/tests/alpaka/CMakeLists.txt
+++ b/tests/alpaka/CMakeLists.txt
@@ -8,5 +8,6 @@ traccc_add_test( alpaka
    alpaka_main.cpp
    LINK_LIBRARIES
    GTest::gtest
-   alpaka
+   GTest::gtest_main
+   alpaka::alpaka
 )
diff --git a/tests/alpaka/alpaka_main.cpp b/tests/alpaka/alpaka_main.cpp
index 2997745..78fbd4f 100644
--- a/tests/alpaka/alpaka_main.cpp
+++ b/tests/alpaka/alpaka_main.cpp
@@ -89,9 +89,3 @@ GTEST_TEST(AlpakaBasic, VectorOp) {
         ASSERT_TRUE(testPassed);
     }
 }
-
-int main(int argc, char** argv) {
-    testing::InitGoogleTest(&argc, argv);
-    int r = RUN_ALL_TESTS();
-    return r;
-}
```